src/Raven.Server/ServerWide/ShardingStore.cs

```diff
@@ -125,9 +133,25 @@ public string ReadPublishedUrls()
                 HttpPooledConnectionLifetime = DocumentConventions.DefaultForServer.HttpPooledConnectionLifetime,
                 UseHttpCompression = _serverStore.Configuration.Sharding.ShardExecutorUseHttpCompression,
                 UseHttpDecompression = _serverStore.Configuration.Sharding.ShardExecutorUseHttpDecompression,
-                GlobalHttpClientTimeout = _serverStore.Configuration.Sharding.OrchestratorTimeoutInMinutes.AsTimeSpan
+                GlobalHttpClientTimeout = _serverStore.Configuration.Sharding.OrchestratorTimeoutInMinutes.AsTimeSpan,
+                CreateHttpClient = handler =>
+                {
+                    handler.ServerCertificateCustomValidationCallback = ShardingCustomValidationCallback;
+                    return new HttpClient(handler);
+                }
             };
 
+        private bool ShardingCustomValidationCallback(HttpRequestMessage message, X509Certificate2 cert, X509Chain chain, SslPolicyErrors errors)
+        {
+            foreach (var urls in _serverStore.PublishedUrls.Urls)
```

Comment:
I can compare this to my own, the problem with that is if each server has different cert it will not work.
Each server can also store his public key(or the thumbprint?) in the state machine and compare that..

Reply:
That is a bad idea, because it does not deal with the cert at all. you can MITM that

See how we do that with TCP connections. The `/tcp/info` is sending back the certificate that will be used, and we compare those.

Note that storing the thumbprint/ etc is not sufficient, what happens when you update the certificate?

In that case, I would rather just have that on the author of the certificate, they are responsible for making this work, instead.